### ISSUE TITLE
`jq` returns unexpected results when sorting JSON by `id` field which is a big interger**

### ISSUE THREAD

**Issue by linnv** [jqlang/jq#3172]
When using `jq` (version 1.6) to sort JSON data by the `id` field, the result is unexpected, with all IDs being the same. However, when performing the same operation using `gojq`, the sorting behaves correctly, returning distinct IDs in the correct order.

**Steps to Reproduce:**

1. Input JSON (`1.json`):

    ```json
    {"id":1725884862351626139}
    {"id":1725884862351626140}
    {"id":1725884862351626138}
    ```

2. Run the following `jq` command:

    ```bash
    jq -c -s 'sort_by(.id)[]' 1.json
    ```

3. Output from `jq`:

    ```json
    {"id":1725884862351626200}
    {"id":1725884862351626200}
    {"id":1725884862351626200}
    ```

4. Compare with the `gojq` command:

    ```bash
    gojq -c -s 'sort_by(.id)[]' 1.json
    ```

5. Correct output from `gojq`:

    ```json
    {"id":1725884862351626138}
    {"id":1725884862351626139}
    {"id":1725884862351626140}
    ```

**Expected Behavior:**
`jq` should return the same result as `gojq`, with the `id` fields correctly sorted in ascending order:

```json
{"id":1725884862351626138}
{"id":1725884862351626139}
{"id":1725884862351626140}
```

**Actual Behavior:**
`jq` returns incorrect results where all `id` fields are set to `1725884862351626200`.

**Versions:**
- `jq`: `jq-1.6`
- `gojq`: `gojq 0.12.16 (rev: 0607aa5/go1.22.3)`

**Environment:**
- OS: 3.10.0-1062.9.1.el7.x86_64 #1 SMP Fri Dec 6 15:49:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux 
- jq installation method: [yum]

**Additional Information:**
The issue seems to be related to `jq-1.6` specifically. Sorting the JSON works as expected with `gojq`. 

**operation log**
```
nce-work@devhost /d/a/t/onelog> jq -c -s 'sort_by(.id)[]' 1.json
{"id":1725884862351626200}
{"id":1725884862351626200}
{"id":1725884862351626200}
nce-work@devhost /d/a/t/onelog> cat 1.json
{"id":1725884862351626139}
{"id":1725884862351626140}
{"id":1725884862351626138}
nce-work@devhost /d/a/t/onelog> jq -c -s 'sort_by(.id)[]' 1.json
{"id":1725884862351626200}
{"id":1725884862351626200}
{"id":1725884862351626200}
nce-work@devhost /d/a/t/onelog> gojq -c -s 'sort_by(.id)[]' 1.json
{"id":1725884862351626138}
{"id":1725884862351626139}
{"id":1725884862351626140}
nce-work@devhost /d/a/t/onelog> jq --version
jq-1.6
nce-work@devhost /d/a/t/onelog> gojq --version
gojq 0.12.16 (rev: 0607aa5/go1.22.3)
```

**Comment by itchyny** [comment]
The behavior was fixed in [jq 1.7](https://github.com/jqlang/jq/releases/tag/jq-1.7). Please use jq 1.7 or newer.

**Comment by linnv** [comment]
Thank you! It works as intended.
